torchrl/envs/model_based/common.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from copy import deepcopy
+from typing import Optional, Union, List
+
+import numpy as np
+import torch
+
+from torchrl.data import TensorDict
+from torchrl.data.utils import DEVICE_TYPING
+from torchrl.envs.common import EnvBase
+from torchrl.modules.tensordict_module import TensorDictModule
+
+__all__ = ["ModelBasedEnv"]
+
+
+class ModelBasedEnv(EnvBase, metaclass=abc.ABCMeta):
+    """
+    Basic environnement for Model Based RL algorithms.
+
+    This class is a wrapper around the model of the MBRL algorithm.
+    This class is meant to give an env framework to a world model (including but not limited to observations, reward, done state and safety constraints models).
+    It is meant to behave as a classical environment.
+
+    This class is meant to be used as a base class for other environments. It is not meant to be used directly.
+
+    An example of use is the following:
+    ```python
```

Comment:
let's write examples using the proper syntax (unless there's a good reason?)
```
"""
Example:
    >>> etc
"""
```

Like this we can test these scripts easily (rather than parsing quoted code snippets)